android/source/VideoLocker/src/org/edx/mobile/base/BaseFragmentActivity.java

```diff
@@ -634,11 +634,15 @@ protected void unblockTouch() {
     }
 
     protected void showWebDialog(String fileName, boolean showTitle, String dialogTitle){
-        WebViewDialogFragment eulaFragment = new WebViewDialogFragment();
-        eulaFragment.setDialogContents(fileName, showTitle, dialogTitle);
-        eulaFragment.show(getSupportFragmentManager(), "dialog");
-        eulaFragment.setStyle(DialogFragment.STYLE_NORMAL,
+        WebViewDialogFragment webViewFragment = new WebViewDialogFragment();
+        webViewFragment.setDialogContents(fileName, showTitle, dialogTitle);
+        webViewFragment.setStyle(DialogFragment.STYLE_NORMAL,
                 android.R.style.Theme_Black_NoTitleBar_Fullscreen);
-        eulaFragment.setCancelable(false);
+        webViewFragment.setCancelable(false);
+        //Show the dialog only if the activity is started. This is to avoid Illegal state
+        //exceptions if the dialog fragment tries to show even if the application is not in foreground
+        if(isActivityStarted()){
```

Comment:
Why isn't this entire method wrapped under `isActivityStarted`?  Is there a reason to create the `webViewFragment` if the application is not in the foreground?